src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -19,7 +19,9 @@ update_state!(model::HydrostaticFreeSurfaceModel) = update_state!(model, model.g
 Update peripheral aspects of the model (auxiliary fields, halo regions, diffusivities,
 hydrostatic pressure) to the current model state.
 """
-function update_state!(model::HydrostaticFreeSurfaceModel, grid)
+update_state!(model::HydrostaticFreeSurfaceModel) = update_state!(model, model.grid)
+
+function update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[])
```

Comment:
This doesn't look right. Shouldn't we have a 3-argument version of `update_state!`? Separating out `model.grid` is how we do some performance optimizations for "single column model mode".

Reply:
damn! sorry... I may have messed up during merging latest main

Reply:
or is it OK?
https://github.com/CliMA/Oceananigans.jl/blob/7f17c1373c8f3ac1d0dbde31f799cda29ce146f0/src/Models/HydrostaticFreeSurfaceModels/single_column_model_mode.jl#L43

Reply:
We need to add the `callbacks=[]` positional argument in `update_state!(model::HydrostaticFreeSurfaceModel, grid::SingleColumnGrid)`, right?

Reply:
and also replace the line you are referring to with `update_state!(model::HydrostaticFreeSurfaceModel, grid::AbstractGrid, callbacks=[])

Reply:
I'm a bit confused because in `main` at the moment I see:

Reply:
I think I fixed it.

Reply:
Yes this is wrong.

Reply:
It should be

```julia
 update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]) = update_state!(model, model.grid, callbacks) 
```

Then

```julia
 function update_state!(model::HydrostaticFreeSurfaceModel, grid, callbacks) 
```

and for the single column grid

```julia
 function update_state!(model::HydrostaticFreeSurfaceModel, grid::SingleColumnGrid, callbacks) 
```

Reply:
It's fixed.